### ISSUE TITLE
Charts causing Issues on ODROID C2

### ISSUE THREAD

**Issue by Flole998** [eclipse/smarthome#4076]
With Version 2.6.x there are issues with the Charts on ODROID C2 as noticed in openhab/openhab-distro#514 aswell as the [OpenHAB Community](https://community.openhab.org/t/oh2-1-fonts-scrambled/33050/8) . The XChart Library should be upgraded to a newer version, I suggest to the newest Version.

**Comment by kaikreuzer** [comment]
> The XChart Library should be upgraded to a newer version

Can you prove that this is the correct path to fix the issue? As shown in https://github.com/openhab/openhab-distro/issues/514#issuecomment-321232874, the current version seems to work fine and at least is not generally broken,

As already mentioned in the forum, the last commit message says:
“3.x was not used due to increased size without benefit and refactoring.”
Did you check the size issues? Why should it now not matter anymore?

**Comment by Flole998** [comment]
It seems to be causing issues with certain Java versions or architectures (something related to the ODROID C2, would need more time and reports to confirm), for me thats reason enough to upgrade. Still waiting for Feedback from the community to see if that fixes the issue. The newest Library seems to be working fine on the C2 according to a community member, if the old one is broken we can safely assume that upgrading would fix this issue.

As far as the size issue, this is completely nonsense to not upgrade just because it is 2MB larger. We could remove it completely and save 500KB, we would just break the Chart Feature (which is broken for some users now anyways). I don't know why the size ever mattered, to me functionality is important and not size. Do you want a small framework or a working framework?

**Comment by davidgraeff** [comment]
Why did the size increase? Does the new version offer more features that might be useful to expose them to the framework?  

In a similar case I used an external library (moquette) and they decided to add additional dependencies at some point (and the resulting jar grew therefore). I just mocked the dependencies away instead and could safe about a megabyte. 

_Sent from my Lge LG-F700K using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Gerry33** [comment]
Pls. kindly refer to my findings regarding this issue at [https://community.openhab.org/t/oh2-1-fonts-scrambled/33050/10](url)

Result from there: It make so sense to upgrade the XCharts to the latest version as it appears in any XChart- version. This bug is related to ODROID C2 and/or JVM running there.  The same XChart- jar runs perfect on e.g. WINDOWS, but produces these problems on ODROIDs. 

Pls. could somebody test the jar's I've linked in the above post.
Thx 
Gerry

**Comment by kaikreuzer** [comment]
Many thanks for your investigations @Gerry33!
So as I expected, merely upgrading XChart won't help to solve this issue.
I agree that the best way forward is to deal with that issue at the XChart project itself.
As there is nothing to do for us in ESH at the moment, I'll close the issue - once a fixed XChart version is available, we can re-open it.

**Comment by Flole998** [comment]
Closing the issue is not going to fix the Problem. If an issue is closed, it means it is resolved or not an issue. This is clearly an issue and also clearly not resolved and should therefor not be closed. If someone is looking at what needs to be done because they want to contribute, they are unable to see this because it is already closed (event though the issue is still there). Another thing we could do is downgrade to the old XChart, that would also solve the issue (the upgrade had no real purpose as far as I can see).

**Comment by Flole998** [comment]
I tested with version 2.2.1 and reverting back would indeed fix the issue. I suggest doing so for now (as 3.x doesn't seem to be an option either as it's not working aswell and also for some reason the 2MB extra are too large for you guys.....). I will create an Issue in the XChart Repository now, even though it doesn't matter for us at the moment.

**Comment by Flole998** [comment]
@kaikreuzer I proposed a downgrade of the XChart library (so basically roll back that PR that created all these issues) and haven't gotten a response yet. This issue has been closed (which I don't understand as it is clearly not resolved yet) so nobody is going to work on it as it doesn't show up anymore. I kindly ask you to reopen this, or do the rollback to the older XChart library (which was working fine before so we can use it) so this can be considered fixed.
